### PR TITLE
boards/nucleo: share board.h and board.c

### PR DESCRIPTION
**Pull request by haukepetersen**
~~Based on #8058~~ and cuts out part from #8044

As #8044 is getting out of hand, I think it makes sense to split its aspects into separate PRs. This PR does two things:
- make a distinction between `common/nucleo` and `common/nucleo64`
- make all nucleo boards use shared `board.h` (by type, one for 32, 64, and 144) and a single shared `board.c` file

Especially the diff of the last commit (`175 additions and 2,293 deletions`) seems to be a good start :-)

### CODE REVIEW

**Comment by haukepetersen** [comment]
rebased, as #8058 was merged.

**Comment by vincent-d** [comment]
Looks good. I have no time to test it today.

Let's give Murdock a try.

**Comment by smlng** [comment]
please also consider #8189 to keep the doc consistent

**Comment by haukepetersen** [comment]
yes, will adapt the doc

**Comment by smlng** [comment]
> yes, will adapt the doc

great! Merging #8189 (before this one), would be nice, too - but it needs approval, first 😉

**Comment by haukepetersen** [comment]
will take a look later today

**Comment by aabadie** [comment]
#8189 is merged, can you rebase @haukepetersen ?

**Comment by haukepetersen** [comment]
sorry for the delay: adapted doxygen groups to the changes in #8189 and rebased.

**Comment by aabadie** [comment]
Looks good for me. I'll test on a subset of boards next week.

File: boards/common/nucleo/board.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *               2016-2017 Inria
+ *               2016-2017 OTA keys
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_nucleo
```
Comment (aabadie) [inline boards/common/nucleo/board.c]: If I get it right, it should be `boards_common_nucleo`

File: boards/common/nucleo/board.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the CPU */
+    cpu_init();
+
+    /* initialization of on-board LEDs
+     * NOTE: LED0 must be explicitly enabled as it is also used for SPI_DEV(0) */
+#ifdef AUTO_INIT_LED0
```
Comment (aabadie) [inline boards/common/nucleo/board.c]: This is only the case for nucleo32 and nucleo64 boards. nucleo144 boards are using different pins for SPI and on-board leds.

**Comment by aabadie** [comment]
Please rebase @haukepetersen

**Comment by haukepetersen** [comment]
rebased and added a fix to always initialize LED0 for Nucleo144 boards.

File: boards/common/nucleo/Makefile.include [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+# all Nucleo boards have an on-board ST-link v2-1 adapter
+export DEBUG_ADAPTER ?= stlink
+export STLINK_VERSION ?= 2-1
+
+# select OpenOCD configuration depending on CPU type
+ifeq (,$(OPENOCD_CONFIG))
+  ifeq ($(CPU),stm32f0)
```
Comment (kaspar030) [inline boards/common/nucleo/Makefile.include]: how about
```export OPENOCD_CONFIG := $(RIOTBOARD)/common/nucleo/dist/$(patsubst stm32%,openocd-%.cfg,$(CPU))```.

And please add an error case, e.g., err out on ```if $(,$(filter stm32%,$(CPU))```

Comment (haukepetersen) [reply boards/common/nucleo/Makefile.include]: sounds good. Ok if I do it in a separate PR?

Comment (haukepetersen) [reply boards/common/nucleo/Makefile.include]: done #8391: https://github.com/RIOT-OS/RIOT/pull/8391/commits/9bbd78d0d8b945b517f459680a5dcd52bb78f546

**Comment by haukepetersen** [comment]
grr, have to rebase again -> because of #8405 this time. Can we please get this merged soon, so this will not happen every time? Also things like #8405 are so much easier to do with this PR merged...

**Comment by haukepetersen** [comment]
adapted to #8405

**Comment by haukepetersen** [comment]
@kaspar030 @aabadie: you think we could this in soonish? I get tired of maintaining this PR and all of your previous comments should be addressed :-)

File: boards/common/nucleo/Makefile.include (hunk 3 above)
Comment (kaspar030) [reply boards/common/nucleo/Makefile.include]: ok

**Comment by kaspar030** [comment]
Tested on nucleo-f401. ACK.

**Comment by haukepetersen** [comment]
finally, thanks a lot for reviewing and testing!

**Pull request merged**